inc/Widget/SmailyWidget.php

```diff
@@ -147,54 +147,307 @@ public function widget( $args, $instance ) {
+							<label for="<?php echo $this->get_field_id( 'smaily_default_background_color' ); ?>" >Use default button color? </label>
+						</div>
+					</div>
+					
+					<div class="column" id="button-text-container">
+						<div>
+							<label for="<?php echo esc_attr( $this->get_field_id( 'button_text_color' ) ); ?>">
+								<b><?php esc_attr_e( 'Button text color', 'smaily' ); ?></b>
+							</label>
+							<!--Jscolor required: false is used to clear the input value. -->
+							<input 
+								data-jscolor="{required:false}"
+								class="button-text-color" type="text" 
+								disabled
+								id="<?php echo esc_attr( $this->get_field_id( 'button_text_color' ) ); ?>"
+								name="<?php echo esc_attr( $this->get_field_name( 'button_text_color' ) ); ?>"
+								value="<?php echo esc_attr( $button_text_color ); ?>" >
+						</div>
+						<div class="default-value-checkbox">
+							<input 
+								class="smaily-checkbox default_text_color" 
+								checked 
+								id="<?php echo $this->get_field_id( 'default_text_color' ); ?>"
+								name="<?php echo $this->get_field_name( 'smaily_default_text_color' ); ?>" 
+								type="checkbox" 
+								value="default_text_color">
+							<label for="<?php echo $this->get_field_id( 'smaily_default_text_color' ); ?>" >Use default text color? </label>
+						</div>
+					</div>
+				</div>
+			</div>
+		</div>
+
+		<script>
+        // Run jQuery in no-conflict mode for WP development.
+        jQuery( document ).ready(function($) {
+
+			// Initate jscolor.install on widget-added.
+			$( document ).on( 'widget-added', function ( event, widget ) {
+				// Load jscolor.
+				jscolor.install();
+            });
+
+            // Initate jscolor.install and input state check on widget update(save.
+            $( document ).on( 'widget-updated', function ( event, widget ) {
+				// Load jscolor.
+                jscolor.install();
+				
+				// If button color is not empty and default color is not checked -> then input isn`t disabled.
+				if ( "<?php echo $button_color; ?>" != "" && "<?php echo $smaily_default_background_color; ?>" == "" ) {
+					$( 'div#button-color-container input[type="text"]' ).prop( "disabled", false );
+					$( 'div#button-color-container input[type="checkbox"]' ).prop( "checked", false );
+				} else {
+					$( 'div#button-color-container input[type="text"]' ).prop( "disabled", true );
+					$( 'div#button-color-container input[type="checkbox"]' ).prop( "checked", true );
+				} 
+				
+				if ( "<?php echo $button_text_color; ?>" != "" && "<?php echo $smaily_default_text_color; ?>" == "" ) {
+					$( 'div#button-text-container input[type="text"]' ).prop( "disabled", false );
+					$( 'div#button-text-container input[type="checkbox"]' ).prop( "checked", false );
+				} else {
+					$( 'div#button-text-container input[type="text"]' ).prop( "disabled", true );
+					$( 'div#button-text-container input[type="checkbox"]' ).prop( "checked", true );
 				}
```

Comment:
This would mean that initial color should be set on widget add, otherwise if user unchecks default color and saves without filling in button color, there is no restriction saving an empty value.

Reply:
True, there are no restrictions in place for saving an empty button color value, but this should not be handled by the form display/front-end logic.

When user tries to save an empty button color value, `SmailyWidget::update` method should have the logic in place to handle the empty value by setting the `smaily_default_background_color` to `default_background_color`.